tests/test_schema.py

```diff
@@ -410,6 +415,10 @@ def testPickle(self):
         schema.addField("i", type="I", doc="int")
         schema.addField("f", type="F", doc="float", units="m2")
         schema.addField("flag", type="Flag", doc="a flag")
+        schema.addField("string", type="String", doc="A string field", size=42)
+        schema.addField("variable_string", type="String", doc="A variable-length string field", size=0)
+        schema.addField("array", type="ArrayF", doc="An array field", size=10)
+        schema.addField("variable_array", type="ArrayF", doc="A variable-length array field", size=0)
```

Comment:
Lots of code duplication; consider factoring the two schema definitions into their own function (`makeTestSchema`? `addCommonTestFields`?).